source-kinesis/capture_test.go

```diff
@@ -193,50 +198,62 @@ func TestKinesisCapture(t *testing.T) {
 	require.NoError(t, err)
 
+	var ctx, cancelFunc = context.WithCancel(context.Background())
+	defer cancelFunc()
 	// Test the discover command and assert that it returns the stream we just created
-	catalog, err := discoverCatalog(configFile)
-	require.NoError(t, err, "discover catalog failed")
+	streamNames, err := listAllStreams(ctx, client);
+  require.NoError(t, err)
 
-	require.GreaterOrEqual(t, len(catalog.Streams), 1)
-	var discoveredStream *airbyte.Stream
-	for _, s := range catalog.Streams {
-		if s.Name == stream {
+	bindings := discoverStreams(ctx, client, streamNames)
+
+	require.GreaterOrEqual(t, len(bindings), 1)
+	var discoveredStream *pc.Response_Discovered_Binding
+	for _, s := range bindings {
+		var res resource
+		err := pf.UnmarshalStrict(s.ResourceConfigJson, &res)
+    require.NoError(t, err);
+
+		if res.Stream == stream {
 			discoveredStream = &s
 			break
 		}
 	}
-	require.NotNil(t, discoveredStream, "missing expected stream")
-	var configuredCatalog = airbyte.ConfiguredCatalog{
-		Streams: []airbyte.ConfiguredStream{{
-			Stream:   *discoveredStream,
-			SyncMode: airbyte.SyncModeIncremental,
-		}},
-		Tail: true,
-	}
-	catalogJson, err := json.Marshal(&configuredCatalog)
+
+	start, err := time.Parse(time.RFC3339Nano, "2022-12-19T14:00:00Z")
 	require.NoError(t, err)
-	var catalogFile = path.Join(tmpDir, "catalog.json")
-	err = ioutil.WriteFile(catalogFile, catalogJson, 0644)
+	end, err := time.Parse(time.RFC3339Nano, "2022-12-19T14:10:00Z")
 	require.NoError(t, err)
 
-	// Test a basic read and assert that we get a proper state message at the end
-	var readArgs = airbyte.ReadCmd{
-		ConfigFile:  configFile,
-		CatalogFile: airbyte.JSONFile(catalogFile),
-		StateFile:   airbyte.JSONFile(stateFile),
-	}
 
-	var ctx, cancelFunc = context.WithCancel(context.Background())
-	defer cancelFunc()
-	var reader, writer = io.Pipe()
+  capture := &st.CaptureSpec{
+		Driver:       new(driver),
+		EndpointSpec: conf,
+		Bindings:     bindings,
+		Validator:    &st.WatchdogValidator{
+			Inner:         &st.ChecksumValidator{},
+			WatchdogTimer: wdt,
+			ResetPeriod:   quiescentTimeout,
+		},
+		Sanitizers:   make(map[string]*regexp.Regexp),
+	}
 
-	go func() {
-		var failure = readStreamsTo(ctx, readArgs, writer)
-		writer.Close()
-		if failure != nil {
-			fmt.Printf("readStreamsTo failed with error: %v\n", failure)
+	// We need to use a relatively long shutdown delay, since the backfill can potentially cover
+	// periods of time with no historical data.
+	const shutdownDelay = 5000 * time.Millisecond
+	var shutdownWatchdog *time.Timer
+
+	capture.Capture(captureCtx, t, func(data json.RawMessage) {
+		if shutdownWatchdog == nil {
+			shutdownWatchdog = time.AfterFunc(shutdownDelay, func() {
+				log.WithField("delay", shutdownDelay.String()).Debug("capture shutdown watchdog expired")
+				cancelCapture()
+			})
 		}
-	}()
+		shutdownWatchdog.Reset(shutdownDelay)
+	})
+	cupaloy.SnapshotT(t, capture.Summary())
```

Comment:
Are these snapshots committed somewhere?

Reply:
@williamhbaker apparently we don't run these tests automatically actually, they are only used to do sanity checks, and it ended up being harder than I expected to make these tests work the way they are written with the new system, so I'm leaving them for now. We can revisit them later.